nncf/quantization/algorithms/algorithm.py

```diff
@@ -26,6 +26,9 @@ class Algorithm(ABC):
     Base class for all Post-Training algorithms.
     """
 
+    def __init__(self):
+        self._algorithm_key = f"BA_{hash(self)}"
+
```

Comment:
This should be implemented as the property.
For example:
```suggestion
    def __init__(self):
        self._algorithm_key = ""

    @property
    def algorithm_key(self) -> str:
        """
        Returns algorithm key.
        
        :return: Algorithm key as string.
        """
        return self._algorithm_key
```
And then use **Algorithm.alrogithm_key** instead of **Algorithm._alrogithm_key**.